hack/generated/pkg/util/armresourceresolver/errors.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package armresourceresolver
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type OwnerNotFound struct {
```

Comment:
I don't think you need this error type. Normally the reason I've seen custom error types used is to enable code to check for them, but it doesn't look like you're doing this other than in tests. They could make sense for standardising error messages too, but this is only raised in one place I think.

Reply:
(Minor though)

Reply:
I'm not checking for it _yet_, but I have a follow-up PR (adding support for ownership in the controller) where I do check for it.

The reason we care about this error is because in the case the specified owner doesn't exist yet we just need to wait further. Any other errors are fatal but this is just a "wait for the owner to exist".

Reply:
Ok, if you're going to start checking for it then cool!